### DISCUSSION TITLE
Google Analytivs 4

### DISCUSSION THREAD

**Discussion by mauthi** [freekmurze/laravel-analytics#394] [Ideas]
Hi there,

I like your package.
Are there any plans to add Google Analytics 4 support to it?

Thx for a short answer.

**Comment by joaowall88** [comment]
Universal Analytics will stop processing new data on standard properties as of July 1, 2023. To prepare, configure and migrate to a Google Analytics 4 property. :(

**Comment by freekmurze** [comment]
I currently have no plans to update this package, but I'm willing to review a PR that makes it compatible with GA4.

**Reply by bartclaeys** [reply]
Kindly asking to update the description of this package to address GA4 so people don't start the installation to then realize it doesn't support GA4?

**Comment by WalrusSoup** [comment]
I could not think of a clean way to implement GA4 inside this same library due to all the changes. For instance, views don't exist anymore. pageViews is now different, metrics are different, etc. So I built one out over [here](https://github.com/MyOutDeskLLC/Laravel-Analytics-V4) using the skeleton package (GOAT template)

**Comment by stefanocurnis** [comment]
I think it could be wise to add a notice on the readme to warn users to avoid installing this in new projects, and that this package won't work anymore in 10 months. I'll submit PR.
edit: or at least to clearly say that it is for Universal Analytics

**Comment by freekmurze** [comment]
We'll update this package soon for GA4 👍

**Reply by gtmassey** [reply]
I just released a package that supports GA4 on PHP8+ and Laravel 9. 

I'm definitely interested in people contributing to the project to support backwards compatibility with older versions of Laravel and PHP.

https://github.com/gtmassey/laravel-analytics

**Reply by JonathanVorich** [reply]
@freekmurze when will the update be available?

**Reply by andyjamo** [reply]
@freekmurze Great to hear you updating the package for GA4. I'm wondering if you have an ETA you are working towards? Also, any considerations for multi-tenanted applications?

**Reply by dayvsonsales** [reply]
Hi, @freekmurze any update about the GA4 support?

**Reply by colt2109** [reply]
Hi, any news concerning GA4 update??

**Reply by Kokil** [reply]
any update about the GA4 support?

**Reply by abzal0** [reply]
Looking for an update with GA4 support as well

**Reply by ferox** [reply]
Looking for an update with GA4 support as well.

**Reply by gotroch** [reply]
+1

**Reply by freekmurze** [reply]
We're actively working on an update of our package. You should see some things hapening in the next few weeks

**Reply by ferox** [reply]
That is awesome!!
Thank you!

**Reply by LuqmanHkim** [reply]
Thank you!

**Reply by Kokil** [reply]
desperately waiting

**Reply by bcbud** [reply]
Keep waiting on that PR to move on over.......... #475

**Comment by Kokil** [comment]
any update about the GA4 support?

**Reply by gtmassey** [reply]
If you need something asap, I wrote a package for it

https://github.com/gtmassey/laravel-analytics

**Reply by ferox** [reply]
Nice work @gtmassey !!!!

Is it compatible with the Spatie package?? 
Would be nice a post migrating from Spatie to your package.

**Reply by ghost** [reply]
@gtmassey 
- today i suggest my team used your package, thank for save my life.

**Reply by gtmassey** [reply]
@ferox 

> Is it compatible with the Spatie package??

Unfortunately it's not a 1-1 replacement, and there might be some class name conflicts if you tried to use both packages at the same time, but as far as GA4 goes, the package should have everything you need to make the queries you want.

If I have some time in the coming weeks I might be able to write a post about migrating between the packages.

**Reply by ferox** [reply]
Thank you, again!!!

**Comment by freekmurze** [comment]
We are ready with the GA4 changes on this package, and are now testing them before releasing a new major release.

Changes have been made on this branch: https://github.com/spatie/laravel-analytics/tree/analytics-api-v4

You can help us test it by requiring `dev-analytics-api-v4` as the package version in your composer.json.

Let me know if this works for you!

**Reply by Kokil** [reply]
I Got  - spatie/laravel-analytics dev-analytics-api-v4 requires laravel/framework ^10.0

as i am using laravel 9

**Reply by freekmurze** [reply]
We will probably stick to requiring Laravel 10, as we can use some specific goodies.

Upgrading a Laravel project from L9 to 10 should be an easy (and recommended) thing to do

**Reply by BrLyes** [reply]
I'm getting

Call to undefined method Spatie\Analytics\Facades\Analytics::isDeferred()

![image](https://user-images.githubusercontent.com/49493569/229144443-63c9e5fd-4ee6-4c8d-924b-14be5a9340f9.png)

**Reply by freekmurze** [reply]
@BrLyes fixed!

**Comment by freekmurze** [comment]
v5.0 of our package, which has support for GA4 has been released 👍

**Reply by mojayaute** [reply]
thanks

**Reply by skibradshaw** [reply]
Will the GA3 Universal Analytics queries still function in V5 of the package?  I'm setting up GA4 now and want to upgrade to V5 to test queries but don't want to lose the data we currently get from the older GA3 Universal Analytics.

**Reply by accexs** [reply]
any reason why it v5 requires laravel 10? i have an app on laravel 9 that requires a lot of work to upgrade to 10

**Reply by AlamTheInnov** [reply]
My application also uses Laravel 8+ so it will be also a lot of work for me, any solution?

**Reply by Kokil** [reply]
Yes, same problem. As to migrate it's a big task as the application is big.
